### PR TITLE
Migrate to no longer use SEP and use astropy Photutils instead.

### PR DESCRIPTION
**Pull request by cmccully**

### CODE REVIEW

File: banzai/photometry.py [hunk 1]
```diff
@@ -23,95 +26,125 @@ def radius_of_contour(contour, source):
...
+
+            sources['flag'] = 0
+
+            # Flag = 1 for sources with bad pixels
+            flag_sources(sources, catalog.labels, deblended_seg_map, image.mask, flag=1, mask_value=1)
+            # Flag = 2 for sources that are deblended
+            flag_deblended(sources, catalog, segmentation_map, deblended_seg_map, flag_value=2)
+            # Flag = 4 for sources that have saturated pixels
+            flag_sources(sources, catalog.labels, deblended_seg_map, image.mask, flag=4, mask_value=2)
+            # Flag = 8 if kron aperture falls off the image
```
Comment (jchate6) [inline banzai/photometry.py:142]: should `flag_edge_sources` be called here?

File: banzai/photometry.py [hunk 2]
```diff
@@ -23,95 +26,125 @@ def radius_of_contour(contour, source):
...
+    sources_off = np.logical_or(sources_off, minor_ymax > ny)
+    sources_off = np.logical_or(sources_off, major_ymax > ny)
+    sources[sources_off] |= flag_value
 
 
 class SourceDetector(Stage):
-    # Note that threshold is number of sigma, not an absolute number because we provide the error
-    # array to SEP.
-    threshold = 10.0
+    threshold = 2.5
```
Comment (jchate6) [inline banzai/photometry.py:71]: where did this threshold come from? Is it no longer a number of sigma but an absolute number?

File: banzai/photometry.py [hunk 3]
```diff
@@ -23,95 +26,125 @@ def radius_of_contour(contour, source):
...
-            sources = Table(sources)
-
-            # We remove anything with a detection flag >= 8
-            # This includes memory overflows and objects that are too close the edge
-            sources = sources[sources['flag'] < 8]
+            segmentation_map = detect_sources(convolved_data, self.threshold, npixels=self.min_area)
+
+            logger.info('Deblending sources', image=image)
+            # Note that nlevels here is DEBLEND_NTHRESH in source extractor which is 32 by default
+            deblended_seg_map = deblend_sources(convolved_data, segmentation_map,
```
Comment (jchate6) [inline banzai/photometry.py:107]: this is the step that kills the memory?

**Review by jchate6 — CHANGES_REQUESTED** [review]
A few comments, but nothing major stands out.

File: banzai/photometry.py (hunk 3 above)
Comment (cmccully) [reply banzai/photometry.py:107]: Yes

File: banzai/photometry.py (hunk 1 above)
Comment (cmccully) [reply banzai/photometry.py:142]: Good catch

**Review by jchate6 — APPROVED** [review]

**Pull request merged**